### DISCUSSION TITLE
Writing up our commitment to expansions and related bug fixes

### DISCUSSION THREAD

**Discussion by kberg** [bafolts/terraforming-mars#2830] [Ideas]
Given the recent discusison about Curiosity II and how it applies to Ares and The Moon,  @nwai90 and I were talking about coming up with an agreement among the developers for how to approach a change like this ongoing.

With that I'd like to make a suggestion for what we write publicly (and similarly what we do internally.) This would be a suggestion of what to write publicly.

> We're building fan expansions, and we'll probably build some more before we're done. While we would love for everything to be perfectly compatible, that's a lot of effort, and possibly a fool's errand. With that in mind, here are the priorites we keep in mind when evaluating how severe a compatibility issue might be.
>
> 1. We endeavor to preserve behavior of all official cards and expansions, and between each other. We will fix bugs that arise due to discovered compatibilty issues, unless they're just demonstrably too much work given our infrastructure. This includes the base game, Corporate Era, Venus, Colonies, Prelude, Turmoil and the Promo cards.
> 
> 2. There are many fan and community expansions, and these may grow. Our goal is to maintain compatibility between a single fan expansion and all the the components listed above, but with less urgency than above. That means fan expansion bugs may linger for some time. This includes (to date) Community cards, Ares, The Moon and Political Agendas.
> 
> 3. Compatibility issues may arise between fan expansions. We'll do our best to make sure things remain compatible between fan expansions, but will leave compatibility gaps, or even change behavior over time to accommodate key changes. That doesn't mean we'll ignore differences between these expansions, but it does mean that we will not endeavor to provide the perfect Every Expansion Experience.
> 
> No matter the issue, this still comes down to being a volunteer effort, a labor of love. Prioritization may be a matter of what feature someone wants to fix. Or what nuanced concern two people want to debate to their deep satisfaction. Words like "endeavor" and "best effort" don't have force behind them, they're just there to give you a sense of our principles when making a decision about getting certain things just right.

**Comment by bafolts** [comment]
What you have here looks great! I am still envious of your writing ability.

I am not sure if it would be related to this commitment but we may want to list these decisions. If some card is not going to be compatible with other cards we may want to start a list of those somewhere. We could reference these if we ever forgot that we agreed to not make a card compatible. Could also potentially tell the user somewhere during game play.

**Reply by kberg** [reply]
> If some card is not going to be compatible with other cards we may want to start a list of those somewhere. 

Perhaps, but let's not put the cart before the horse - I don't think this is that big of a problem yet, just pre-emptive.

**Comment by kevinb9n** [comment]
I'll observe that this seems to conceptualize compatibility as a many-to-many issue - any pair of cards might be compatible or not.

An alternative is to have a set of behaviors/assumptions/invariants which you require each card to be compatible with. When two cards are, there is "no way" for them to not be compatible with each other.  And when growing this model to accommodate a fan card is too hard, why add that fan card then?

Examples (they're all filed already):

* A card like Pets has the established authority to make its own resources non-removable. Thus, a fan card like Project Workshop should have no way to remove them. Conflict resolved.
* First, the core game defines what "playing a card" and "playing a tag" mean. This is usually from-hand, but Valley Trust and Self-Replicating Robots establish that it can happen from a custom location. Secondly, a card like Dirigibles has the established authority to say that when its resources function exactly as if they are 3 MC each. Conclusion, a fan card like Playwrights should have no way to stop that Dirigible from being used. Conflict resolved.
* The simple fact that no Prelude card is Active is arguably reason enough not to allow By-Election. (The reverse could be said of Midas, but you can at least think of it as a degenerate case of an Active card.)
* Which expansion a card belongs to is not game vocabulary (only used for building the deck). There's no card in canon that depends on it in any way, and the developers even felt comfortable using the phrase "a Venus card" to mean a card with a Venus *tag*. Political Uprising breaks this rule, and by saying "a Turmoil card" will even confuse users who next see "a Venus card". (Yes, the expansion is called "Venus Next", my I think my points still stand.)

So, a model, and fidelity to the model.

**Reply by kevinb9n** [reply]
Another thing:  at some point, it may become tempting to resolve card conflicts by defining global precedence rules, like "first played wins", "corporation wins", "active player's card wins", etc. (Of course, what a card says about *itself* has a sort of "precedence" over what some other card tries to say about it, e.g. the Dirigibles and Pets examples above).

I'm glad to see no reference to this here, and I think it's worth avoiding taking that step. Nothing in canon establishes anything like it.

A way to prevent conflicts that I haven't mentioned is this: card effects must never contradict core game rules, because as soon as two try to contradict the same game rule in different ways, you're in a pickle. To make that work, a few cards in canon should be fixed, but the fixes are both-ways compatible: Ecoline (1 plant less), Polyphemos (2 MC more), Terralabs (2 MC less), Lakefront (it gives 1 MC, not 3).

**Reply by kberg** [reply]
This is a good idea, and can be considered as a further refinement or replacement. But for now this solves some pressing discussion points.

**Comment by kberg** [comment]
Given that the general idea hasn't met much in the way of resistance (modulo Kevin's reasoned ideas) I'm going to move this into the Wiki.

**Reply by kberg** [reply]
(no resistance and three thumbs up.)

**Comment by kberg** [comment]
This is done.
